parsing/src/parts.ts

```diff
@@ -183,17 +204,21 @@ function getItems(data: any[]): PartDataInterface {
                     .find((entry: any) => entry.ClassName === `Desc_${productClass}_C`);
 
                 // Extract stack size
-                const stackSize = stackSizeConvert(classDescriptor?.mStackSize || "SS_UNKNOWN")
+                const stackSize: number = stackSizeConvert(classDescriptor?.mStackSize || "SS_UNKNOWN")
+                // Extract the energy value
+                const energyValue: number = classDescriptor?.mEnergyValue;
 
                 // Check if the part is a collectable (e.g., Power Slug)
                 if (isCollectable(entry.mIngredients)) {
                     collectables[partName] = friendlyName;
                 } else {
+                    //console.log(`Adding part: ${partName} (${friendlyName}) with energy value: ${energyValue}`);
                     parts[partName] = {
                         name: friendlyName,
                         stackSize,
                         isFluid: isFluid(partName),
                         isFicsmas: isFicsmas(entry.mDisplayName),
+                        energyGeneratedInMJ: Math.round(energyValue), // Round to the nearest whole number (all energy numbers are whole numbers) 
```

Comment:
What happens if it's undefined (see line 209)? Math.round on a undefined value sounds.. dangerous

Reply:
resolved on 216.